### ISSUE TITLE
TERRAFORM_PLAN_HAS_CHANGES not being set/not working

### ISSUE THREAD

**Issue by MatiasNielsen** [charleszipp/azure-pipelines-tasks-terraform#371]
**Describe the bug**
Trying to skip a stage based on TERRAFORM_PLAN_HAS_CHANGES isn't working correctly

**To Reproduce**

According to the documentation, plan has to be run with detailed-exitcode.

Run terraform-plan with -detailed-exitcode:
```
- task: TerraformCLI@0
  displayName: 'terraform plan'
  inputs:
    command: plan
    environmentServiceName: 'My Azure Service Connection'
    commandOptions: '-out=$(System.DefaultWorkingDirectory)/terraform.tfplan -detailed-exitcode'
```
Running with apply afterwards:

```
 - stage: POGodkendelse
    dependsOn:
      - Statisk_kodeanalyse
      - validate
    condition: and(succeeded(), eq(variables['TERRAFORM_PLAN_HAS_CHANGES'], 'true'))
```
The stage POGodkendelse is simply skipped.
![image](https://github.com/jason-johnson/azure-pipelines-tasks-terraform/assets/15323190/cf2afc0b-c28a-4208-871a-916d045ca44c)
Even though there is obvious changes:
![image](https://github.com/jason-johnson/azure-pipelines-tasks-terraform/assets/15323190/eba08da4-9b91-4cfb-a455-8f609be754f7)

**Expected behavior**
It shouldn't be skipping the pipeline when theres changes..

**Screenshots**
See above

**Agent Configuration**
 - OS: Ubuntu
 - Self Hosted

**Comment by MatiasNielsen** [comment]
I've tried echoing the variable and it seems to be returning true. The condition isn't working though.

**Comment by jason-johnson** [comment]
How do you know this a problem with the extension and not your pipeline?  Have you tried setting a variable by hand (e.g. with bash or powershell) and verifying that the conditional works with that?

**Comment by jason-johnson** [comment]
Oh, I think I see what's going on.  You are running output in one stage and trying to test the condition in another stage.  This is not currently supported as explained in #16.  And even when we fix #16, what you show here will never work.  You have to access variables across stages as shown [here](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#use-outputs-in-a-different-stage).

Looks like this is a duplicate.
